tests/integrationv2/test_client_authentication.py

```diff
@@ -11,116 +11,147 @@
-        client_certificate_file=certificate.cert,
+        key=client_certificate.key,
+        cert=client_certificate.cert,
+        trust_store=certificate.cert,
         insecure=False,
         protocol=protocol)
 
     server_options = copy.copy(client_options)
     server_options.data_to_send = None
     server_options.mode = Provider.ServerMode
-    server_options.key = Certificates.RSA_2048_SHA256_WILDCARD.key
-    server_options.cert = Certificates.RSA_2048_SHA256_WILDCARD.cert
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+    server_options.trust_store = client_certificate.cert
 
-    # Passing the type of client and server as a parameter will
-    # allow us to use a fixture to enumerate all possibilities.
     server = managed_process(S2N, server_options, timeout=5)
     client = managed_process(provider, client_options, timeout=5)
 
-    # The client should connect and return without error
+    # Openssl should send a client certificate and complete the handshake
     for results in client.get_results():
         assert results.exception is None
         assert results.exit_code == 0
+        assert b'write client certificate' in results.stderr
+        assert b'write certificate verify' in results.stderr
+        assert_openssl_handshake_complete(results)
 
-    expected_version = get_expected_s2n_version(protocol, provider)
 
-    # S2N should indicate the procotol version in a successful connection.
+    # S2N should successfully connect
     for results in server.get_results():
         assert results.exception is None
         assert results.exit_code == 0
-        assert bytes("Actual protocol version: {}".format(expected_version).encode('utf-8')) in results.stdout
+        assert_s2n_handshake_complete(results, protocol, provider)
         assert random_bytes in results.stdout
 
 
 @pytest.mark.uncollect_if(func=invalid_test_parameters)
-@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
-@pytest.mark.parametrize("provider", [OpenSSL])
-@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL], ids=get_parameter_name)
 @pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
-@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
-def test_client_auth_with_s2n_server_using_nonmatching_certs(managed_process, cipher, provider, curve, protocol, certificate):
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", CERTS_TO_TEST, ids=get_parameter_name)
+@pytest.mark.parametrize("client_certificate", CERTS_TO_TEST, ids=get_parameter_name)
+def test_client_auth_with_s2n_server_using_nonmatching_certs(managed_process, cipher, provider, protocol, certificate, client_certificate):
     port = next(available_ports)
 
+    if protocol < Protocols.TLS12 and client_certificate.algorithm == 'EC':
```

Comment:
Nit: Why are we calling the signature algorithm`EC` instead of ` ECDSA`? Seems misleading.

Reply:
We unfortunately use "EC" all over the integ tests. I'm planning to change it in a later PR, but I'd argue it's out of scope of this fix.